File: c#/guess-number.cs
using System;

namespace GuessNumberGame
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("=====================================");
            Console.WriteLine("     欢迎来到猜数字游戏！");
            Console.WriteLine("=====================================");
            Console.WriteLine("游戏规则：");
            Console.WriteLine("1. 系统会随机生成一个1-100之间的数字");
            Console.WriteLine("2. 你有10次机会猜中这个数字");
            Console.WriteLine("3. 每次猜测后会提示'大了'或'小了'");
            Console.WriteLine("=====================================\n");
            
            bool playAgain = true;
            while (playAgain)
            {
                PlayGame();
                
                Console.Write("\n是否再来一局？(y/n): ");
                string input = Console.ReadLine();
                playAgain = input.ToLower() == "y" || input.ToLower() == "yes";
                Console.WriteLine();
            }
            Console.WriteLine("感谢游玩，再见！");
        }
        
        static void PlayGame()
        {
            Random random = new Random();
            int targetNumber = random.Next(1, 101); // 生成1-100之间的随机数
            int maxAttempts = 10;
            int attempts = 0;
            bool isGuessed = false;
            
            Console.WriteLine($"新游戏开始！你有{maxAttempts}次机会。\n");
            
            while (attempts < maxAttempts)
            {
                int remaining = maxAttempts - attempts;
                Console.Write($"请输入你的猜测 (剩余{remaining}次机会): ");
                
                // 获取用户输入并验证
                string input = Console.ReadLine();
                if (!int.TryParse(input, out int guess))
                {
                    Console.WriteLine("请输入有效的数字！\n");
                    continue;
                }
                
                // 验证输入范围
                if (guess < 1 || guess > 100)
                {
                    Console.WriteLine("请输入1-100之间的数字！\n");
                    continue;
                }
                
                attempts++;
                
                // 判断猜测结果
                if (guess == targetNumber)
                {
                    Console.WriteLine($"\n恭喜你猜对了！就是 {targetNumber}！");
                    Console.WriteLine($"你用了 {attempts} 次机会。");
                    isGuessed = true;
                    break;
                }
                else if (guess < targetNumber)
                {
                    Console.WriteLine("太小了！再大一点。\n");
                }
                else
                {
                    Console.WriteLine("太大了！再小一点。\n");
                }
            }
            
            // 如果10次都没猜中
            if (!isGuessed)
            {
                Console.WriteLine($"\n很遗憾，10次机会用完了！");
                Console.WriteLine($"正确答案是：{targetNumber}");
            }
            
            // 显示统计信息
            Console.WriteLine($"\n本局使用了 {attempts} 次机会。");
        }
    }
}

File: java/guess-number.java
import java.util.Random;
import java.util.Scanner;

public class GuessNumberGame {
    
    public static void main(String[] args) {
        // 创建随机数生成器和扫描器
        Random random = new Random();
        Scanner scanner = new Scanner(System.in);
        
        // 生成1-100的随机数
        int targetNumber = random.nextInt(100) + 1;
        int maxAttempts = 10;
        int attempts = 0;
        boolean hasGuessedCorrectly = false;
        
        System.out.println("========================================");
        System.out.println("        欢迎来到猜数字游戏！");
        System.out.println("========================================");
        System.out.println("游戏规则：");
        System.out.println("1. 系统已随机生成一个 1-100 之间的整数");
        System.out.println("2. 您共有 " + maxAttempts + " 次猜测机会");
        System.out.println("3. 每次猜测后，系统会提示您猜大或猜小");
        System.out.println("========================================\n");
        
        // 游戏主循环
        while (attempts < maxAttempts) {
            int remainingAttempts = maxAttempts - attempts;
            System.out.print("请输入您的猜测（剩余 " + remainingAttempts + " 次机会）: ");
            
            // 获取玩家输入并验证
            int guess;
            try {
                guess = scanner.nextInt();
            } catch (Exception e) {
                System.out.println("请输入有效的整数！");
                scanner.next(); // 清除无效输入
                continue;
            }
            
            // 验证输入范围
            if (guess < 1 || guess > 100) {
                System.out.println("请输入 1-100 之间的数字！");
                continue;
            }
            
            attempts++;
            
            // 比较猜测值与目标值
            if (guess == targetNumber) {
                hasGuessedCorrectly = true;
                System.out.println("\n恭喜您！猜对了！");
                System.out.println("目标数字就是 " + targetNumber);
                System.out.println("您用了 " + attempts + " 次机会猜中！");
                break;
            } else if (guess < targetNumber) {
                System.out.println("猜小了！再大一点");
            } else {
                System.out.println("猜大了！再小一点");
            }
            
            // 提示剩余机会
            if (attempts < maxAttempts) {
                System.out.println("----------------------------------------\n");
            }
        }
        
        // 游戏结束处理
        if (!hasGuessedCorrectly) {
            System.out.println("\n========================================");
            System.out.println("很遗憾，您用完了所有机会！");
            System.out.println("正确的数字是： " + targetNumber);
            System.out.println("========================================");
        } else {
            System.out.println("========================================");
            System.out.println("          游戏结束，感谢参与！");
            System.out.println("========================================");
        }
        
        scanner.close();
    }
}